btree/btree.c

```diff
@@ -3865,19 +3879,27 @@ static int64_t btree_put_kv_tick(struct m0_sm_op *smop)
 			}
 		}
 		/** Fall through if path_check is successful. */
-	case P_MAKESPACE: {
-		if (oi->i_key_found) {
-			struct m0_btree_rec rec;
-			rec.r_flags = M0_BSC_KEY_EXISTS;
-			int rc = bop->bo_cb.c_act(&bop->bo_cb, &rec);
-			if (rc) {
-				lock_op_unlock(tree);
-				return fail(bop, rc);
-			}
+	case P_SANITY_CHECK: {
+		int  rc;
+		if (bop->bo_opc == M0_BO_PUT && oi->i_key_found)
+			rc = M0_BSC_KEY_EXISTS;
+		else if (bop->bo_opc == M0_BO_UPDATE && !oi->i_key_found)
+			rc = M0_BSC_KEY_NOT_FOUND;
+		else
+			rc = M0_BSC_SUCCESS;
+
+		if (rc != M0_BSC_SUCCESS) {
+			oi->i_sm_rc = rc;
 			lock_op_unlock(tree);
-			return m0_sm_op_sub(&bop->bo_op, P_CLEANUP, P_FINI);
+			return fail(bop, M0_ERR(-rc));
```

Comment:
`enum m0_btree_status_codes` return codes should not be used for returning tick machine return codes. We are using this enum as flag in record. State machine return codes should be linux codes.

Reply:
okay, I am using m0_btree_status_codes as return codes , as it will help debugging. but if it is necessary to s=use linux codes for return code I will update.